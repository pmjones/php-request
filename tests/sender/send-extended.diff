001+ Fatal error: Uncaught Error: Call to undefined method ExtResponse::getHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php:20
002+ Stack trace:
003+ #0 /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php(11): SapiResponseSender->registerHeaderCallback(Object(ExtResponse))
004+ #1 /Users/pmjones/Code/pmjones/php-request/tests/sender/send-extended.php(9): SapiResponseSender->send(Object(ExtResponse))
005+ #2 {main}
006+   thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php on line 20
001- content
002- array(3) {
003-   [0]=>
004-   string(8) "foo: bar"
005-   [1]=>
006-   string(19) "Set-Cookie: baz=dib"
007-   [2]=>
008-   string(38) "Content-type: text/html; charset=UTF-8"
009- }