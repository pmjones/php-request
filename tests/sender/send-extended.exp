content
array(3) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(19) "Set-Cookie: baz=dib"
  [2]=>
  string(38) "Content-type: text/html; charset=UTF-8"
}