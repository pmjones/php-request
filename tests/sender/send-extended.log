
---- EXPECTED OUTPUT
content
array(3) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(19) "Set-Cookie: baz=dib"
  [2]=>
  string(38) "Content-type: text/html; charset=UTF-8"
}
---- ACTUAL OUTPUT
Fatal error: Uncaught Error: Call to undefined method ExtResponse::getHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php:20
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php(11): SapiResponseSender->registerHeaderCallback(Object(ExtResponse))
#1 /Users/pmjones/Code/pmjones/php-request/tests/sender/send-extended.php(9): SapiResponseSender->send(Object(ExtResponse))
#2 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php on line 20
---- FAILED
