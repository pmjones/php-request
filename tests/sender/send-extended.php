<?php
class ExtResponse extends SapiResponse
{
}
$ext = new ExtResponse();
$ext->setHeader('foo', 'bar');
$ext->setCookie('baz', 'dib');
$ext->setContent('content');
(new SapiResponseSender())->send($ext);
echo "\n";
var_dump(headers_list());
