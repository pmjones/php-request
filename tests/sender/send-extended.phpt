--TEST--
SapiResponseSender::send (extended without constructor)
--CGI--
--INI--
expose_php=0
--FILE--
<?php
class ExtResponse extends SapiResponse
{
}
$ext = new ExtResponse();
$ext->setHeader('foo', 'bar');
$ext->setCookie('baz', 'dib');
$ext->setContent('content');
(new SapiResponseSender())->send($ext);
echo "\n";
var_dump(headers_list());
--EXPECT--
content
array(3) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(19) "Set-Cookie: baz=dib"
  [2]=>
  string(38) "Content-type: text/html; charset=UTF-8"
}
