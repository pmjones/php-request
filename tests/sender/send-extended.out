Fatal error: Uncaught Error: Call to undefined method ExtResponse::getHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php:20
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php(11): SapiResponseSender->registerHeaderCallback(Object(ExtResponse))
#1 /Users/pmjones/Code/pmjones/php-request/tests/sender/send-extended.php(9): SapiResponseSender->send(Object(ExtResponse))
#2 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php on line 20